src/ArrayHelper.php

```diff
@@ -378,8 +378,163 @@ public static function setValue(array &$array, array|float|int|string|null $key,
+     *           'in' => [
+     *               'variable1'
+     *               'variable2'
+     *           ]
+     *       ]
+     *   ]
+     *  ```
+     *
+     *   If the value exists, it will become the first element of the array
+     *   ```php
+     *       $array = [
+     *           'key' => 'in',
+     *       ];
+     *   ```
+     *
+     *   The result of:
+     *   ```php
+     *       ArrayHelper::addValue($array, 'key.in', 'variable1');
+     *    ```
+     *
+     *   will be the following:
+     *   ```php
+     *       [
+     *           'key' => ['in', 'variable1'],
+     *       ]
+     *  ```
+     *
+     * @param array $array The array to write the value to.
+     * @param array|float|int|string|null $key The path of where do you want to write a value to `$array`
+     *  the path can be described by an array of keys. If the path is null then `$array` will be assigned the `[$value]`.
+     *
+     * @psalm-param ArrayKey|null $key
+     *
+     * @param mixed $value The value to be written.
+     */
+    public static function addValue(array &$array, array|float|int|string|null $key, mixed $value): void
+    {
+        if ($key === null) {
+            /** @var mixed */
+            $array [] = $value;
+            return;
+        }
+
+        $keys = is_array($key) ? $key : [$key];
+
+        while (count($keys) > 0) {
+            $k = self::normalizeArrayKey(array_shift($keys));
+            if (!isset($array[$k])) {
+                $array[$k] = [];
+            }
+            if (!is_array($array[$k])) {
+                $array[$k] = [$array[$k]];
+            }
+            $array = &$array[$k];
+        }
+
+        /** @var mixed */
+        $array [] = $value;
+    }
+
     /**
      * Writes a value into an associative array at the key path specified.
+     * If there is no such key path yet, it will be created recursively and an empty array will be initialized.
+     *
+     * ```php
+     *   $array = [
+     *      'key' => [
+     *
+     *        ]
+     *    ];
+     * ```
+     *
+     * The result of:
+     *
+     *  ```php
+     *   ArrayHelper::addValue($array, 'key.in', 'variable1');
+     *   ArrayHelper::addValue($array, 'key.in', 'variable2');
+     *  ```
+     *
+     *   will be the following:
+     *
+     *   ```php
+     *    [
+     *        'key' => [
+     *            'in' => [
+     *                'variable1'
+     *                'variable2'
+     *            ]
+     *        ]
+     *    ]
+     *   ```
+     *
+     *   If the value exists, it will become the first element of the array
+     *   ```php
+     *       $array = [
+     *           'key' => 'in',
+     *       ];
+     *   ```
+     *
+     *   The result of:
+     *   ```php
+     *       ArrayHelper::addValue($array, 'key.in', 'variable1');
+     *    ```
+     *
+     *   will be the following:
+     *   ```php
+     *       [
+     *           'key' => ['in', 'variable1'],
+     *       ]
```

Comment:
```suggestion
     * [
     *     'key' => [
     *         'in',
     *         'in' => 'variable1',
     *     ],
     * ]
```

Is this right behavior that key `in` from `key.in` becomes a value?